apps/src/JSInterpreter.js

```diff
@@ -475,8 +487,12 @@ JSInterpreter.prototype.getProperty = function (obj, name) {
 JSInterpreter.prototype.setProperty = function(obj, name, value,
                                                opt_fixed, opt_nonenum) {
   name = name.toString();
+  var nativeParent;
   if (obj.isCustomMarshal) {
     obj.data[name] = codegen.marshalInterpreterToNative(this.interpreter, value);
+  } else if (obj === this.globalScope &&
+      (!!(nativeParent = this.customMarshalGlobalProperties[name]))) {
```

Comment:
This expression/assignment is duplicated earlier in the function, line 457.  Possible extraction?